keylime/keylime_agent.py

```diff
@@ -627,6 +637,20 @@ def main():
             raise RuntimeError('agent_uuid is configured to use dmidecode, '
                                'but it\'s is not found on the system.')
 
+    # initialize the tmpfs partition to store keys if it isn't already available
+    secdir = secure_mount.mount()
+
+    # Now that root-owned files are opened drop privileged
+    # If 'run_as' is not available keep running with high privileges
+    if os.getuid() == 0:
```

Comment:
Just a thought: I wonder if we should create an additional setting, e.g `run_as_root` to be false by default, and only run the agent as root if this is explicitly enabled/asked for by the configuration.

Reply:
I think that's the same as having the configuration `run_as = foo:bar` set to some explicit values, such as `foo:bar`. I left it blank in the default config file for backwards compatibility purposes.

Reply:
We should switch to `keylime:tss` in a future release, but only with a major version change. The current default is backwards compatible to current installs.

Reply:
While running as non-root by default would be better from a security perspective, I am afraid that such settings would not allow fluent upgrade for existing users. We should think through and document what would be the recommended setup and whether it could be automated (including the migration). 
One thing that comes to my mind is using certificates provided by CV (cacert+cline-cert) which is the current default. Either we change the default or find a way how to make these certificates accessible to the agent. Also, agent should not be able write to `/var/lib/keylime` directly (I am looking at you `tpmdata.yml`) as there is also the content of CV and R and enabling write also means we are enabling removal. Therefore agent should have a dedicated directory for all its content owned by the agent. If CV is configured to create these certificates it could prepare everything for the agent. There could be a dedicated script doing that so an administrator could eventually reuse it.

Reply:
Also, I think that this topic would probably deserve dedicated discussion on the keylime channel.